README.md

```diff
@@ -78,103 +101,366 @@ Modify the app's **android/app/src/main/AndroidManifest.xml** to include the fol
-| version | false | "" | The version of YOUR application |
-| enableCrashReporting | false | false | Enables the usage of Raygun's Crash Reporter |
-| disableNativeCrashReporting | false | false | Given you have crash reporting enabled, this disables Crash Reporting from reporting on errors that occur outside the scope of YOUR program |
-| enableRealUserMonitoring | false | false | Enables the usage of Raygun's Real User Monitoring |
-| disableNetworkMonitoring | false | false | Given you have real user monitoring enabled, this disables the monitor from recording statistics based on Network usage |
-| customCrashReportingEndpoint | false | "" | Sends all Crash Reporting information to an alternative address |
-| customRealUserMonitoringEndpoint | false | "" | Sends all Real User Monitoring information to an alternative address |
-| onBeforeSendingCrashReport | false | null | A handler for crash reports before they are sent (more below) |
-| ignoredURLs | false | [] | An array of URLs to ignore. This is utilized with network monitoring, which will be useless without enabling Real User Monitoring (and leaving Network Monitoring enabled) |
+###  Breadcrumb 
+A Breadcrumb is a container for simple pieces of information that are used to aid with identifying
+issues. These are sent away with a [CrashReportPayload](#crashreportpayload).
 
-It looks a little something like:
+```typescript
+export type Breadcrumb = {
+  message: string;
+  category?: string;
+  level?: 'debug' | 'info' | 'warning' | 'error';
+  customData?: CustomData; // See more on CustomData below
+  timestamp?: number;
+};
+```
+
+<br/>
+<br/>
 
+###  BreadcrumbOption 
+The BreadcrumbOption is similar to the Breadcrumb (see above) however it only has three fields.
+`category`, `level`, and `customData` are the only fields used in this object type. This object
+type gets used as a parameter for [recordBreadcrumb](#rec)
 ```typescript
-export type RaygunClientOptions = {
-  apiKey?: string;
-  version?: string;
-  enableCrashReporting?: boolean;
-  disableNativeCrashReporting?: boolean;
-  enableRealUserMonitoring?: boolean;
-  disableNetworkMonitoring?: boolean;
-  customCrashReportingEndpoint?: string;
-  customRealUserMonitoringEndpoint?: string;
-  onBeforeSendingCrashReport?: BeforeSendHandler;
-  ignoredURLs?: string[];
-}
+export type BreadcrumbOption = Omit<Breadcrumb, 'message' | 'timestamp'>;
 ```
 
-Note: All parameters in the RaygunClientOptions are NOT required.
-Any parameter that is not specified will be set to their default / empty values (based on their
-type). Thus, you can initialize a client with the like the following example shows:
+<br/>
+<br/>
+
+###  CrashReportPayload 
+This object is only accessible within the [BeforeSendHandler](#beforesendhandler) function. 
+This reference should aid in designing an intermediate function. It acts as a container for basic 
+information about some crash.
```

Comment:
_Minor:_ `information about some crash` or `information about a crash`

Also I wonder if we should be using the term `Crash`, `Error` or `Exception`

Reply:
I talk about a crash as I feel it encapsulates both Errors and Exceptions.... however I do remember talking to Mitch the other day about this. TS uses Exceptions, so I can go through and reword these if you think using technically accurate terms here would benefit the customer (or would it confused them further?)

IN PROGRESS